metaflow/datastore/content_addressed_store.py

```diff
@@ -145,38 +138,49 @@ def load_blobs(self, keys, force_raw=False):
         else:
             to_load = keys
         to_load_paths = [
-            self._backend.path_join(
-                self._prefix, k[:2] if not k.startswith('r_') else k[2:4], k)
-            for k in to_load]
+            self._backend.path_join(self._prefix, k[:2], k) for k in to_load]
         load_results = self._backend.load_bytes(to_load_paths)
         for k, path in zip(to_load, to_load_paths):
             # At this point, we either return the object as is (if raw) or
             # decode it according to the encoding version
-            result = load_results[path]
-            if force_raw or k.startswith('r_'):
+            result, meta = load_results[path]
+            if force_raw or (meta and meta.get('cas_raw', False)):
                 with result as r:
                     results[k] = r.read()
             else:
                 with result as r:
-                    magic, version = struct.unpack(
-                        self._info_format,
-                        r.read(struct.calcsize(self._info_format)))
-                    try:
-                        # We need to see if this is an artifact saved by this
-                        # version (or newer) of the datastore. Prior to this,
-                        # things were saved gzipped but with no magic string.
-                        if magic == self._magic:
-                            unpack_code = getattr(self, '_unpack_v%d' % version)
+                    if meta is None:
+                        # Backward compatible mode
+                        # We also need to deal with a short-lived backward
+                        # compatibility mode where the version was encoded
+                        # in the file itself
+                        magic = b'.MFBlob'
+                        # Pack format for header of non-raw blobs.
+                        # The integer corresponds to the version number of the encoding
+                        info_format = '<%dsi' % len(magic)
+                        file_magic, version = struct.unpack(
+                            info_format,
+                            r.read(struct.calcsize(info_format)))
+                        if file_magic == magic:
+                            unpack_code = getattr(
+                                self, '_unpack_v%d' % version, None)
                         else:
-                            # This is the backward compatible mode
                             r.seek(0)
+                            # This is the backward compatible mode
                             unpack_code = getattr(
```

Comment:
why not just ```
unpack_code = self._unpack_backward_compatible
```

Reply:
Yes, dumb copy paste without thinking...

Reply:
Done and simplified.